src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -206,47 +272,16 @@ impl InserterInnerBuilder {
             .send()
             .await
             .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
-        // TODO: shall we let `reqwest` handle the redirect?
-        let mut be_url = if resp.status() == StatusCode::TEMPORARY_REDIRECT {
-            resp.headers()
-                .get("location")
-                .ok_or_else(|| {
-                    SinkError::DorisStarrocksConnect(anyhow::anyhow!(
-                        "Can't get doris BE url in header",
-                    ))
-                })?
-                .to_str()
-                .context("Can't get doris BE url in header")
-                .map_err(SinkError::DorisStarrocksConnect)?
-                .to_string()
-        } else {
-            return Err(SinkError::DorisStarrocksConnect(anyhow::anyhow!(
-                "Can't get doris BE url",
-            )));
-        };
 
-        if self.fe_host != LOCALHOST && self.fe_host != LOCALHOST_IP {
-            let mut parsed_be_url =
-                Url::parse(&be_url).map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
-            let be_host = parsed_be_url.host_str().ok_or_else(|| {
-                SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't get be host from url"))
-            })?;
-
-            if be_host == LOCALHOST || be_host == LOCALHOST_IP {
-                // if be host is 127.0.0.1, we may can't connect to it directly,
-                // so replace it with fe host
-                parsed_be_url
-                    .set_host(Some(self.fe_host.as_str()))
-                    .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
-                be_url = parsed_be_url.as_str().into();
-            }
-        }
+        let be_url = try_get_be_url(&resp, self.fe_host.clone())?.ok_or_else(|| {
+            SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't get doris BE url",))
+        })?;
```

Comment:
Should use the error returned by `try_get_be_url`, rather than `anyhow!`

```suggestion
        let be_url = try_get_be_url(&resp, self.fe_host.clone())?;
```

Reply:
You're right, the returned error should not be wrapped twice.

Reply:
And I don't find any documents in starrocks that describe BE redirection behavior. we cannot make sure in which situation the FE should redirect to BE or not.

Reply:
BTW, if `reqwest` supports auto redirection without removing sensitive headers like `Authorization`, we may make our code more elegant.